### PR TITLE
Implement isUnlocked, unlock, handled INCREMENT, and CHECK_STORY

### PR DESCRIPTION
**Pull request by asoto01**

### CODE REVIEW

**Comment by asoto01** [comment]
hw1

**Comment by asoto01** [comment]
I wonder why I am not seeing travis. I did pass the test on my computer, when I ran "npm test".
hw1

**Comment by rcliao** [comment]
You are not seeing TravisCI status because you are missing a file called `.travis.yml` under your root directory.

File: lab3/client/src/reducer.js [hunk 1]
```diff
@@ -1,32 +1,56 @@
...
-		return state;
-	case 'INCREMENT_COUNTER':
-		state.counter = action.payload + 1;
 		return state;
+
+		case 'INCREMENT_CLICK':
+			state.counter = action.payload + 1;
+			return state;
+
+		case 'INCREMENT':
```
Comment (rcliao) [inline lab3/client/src/reducer.js:32]: a better name for this would be `INCREMENT_FROM_GENERATORS`

File: lab3/client/src/reducer.js [hunk 2]
```diff
@@ -1,32 +1,56 @@
...
+				state.counter += state.generators[x].rate
+			}
+		}	
+			return state;
+
+		case 'CHECK_STORY':
+			for (var x in state.story) {
+				const checkStory = new Story(Object.assign({}, state.story[x]))
+				if (checkStory.isUnlockYet(state.counter)) {
+					state.story[x].state = checkStory.unlock();
```
Comment (rcliao) [inline lab3/client/src/reducer.js:47]: `unlock()` method doesn't change the state. Are you sure you have update the state correctly?

**Pull request merged**